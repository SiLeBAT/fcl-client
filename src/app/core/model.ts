export interface ValidationError {
    code: string;
    message: string;
}
