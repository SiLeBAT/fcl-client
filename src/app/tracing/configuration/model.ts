import {
    Color,
    DataServiceInputState,
    LabelPart,
    NodeShapeType,
    TableColumn,
    TableRow,
} from "../data.model";
import { ComplexFilterCondition, PropToValuesMap } from "./configuration.model";

export interface RowFilter<T> {
    filter(arr: TableRow[]): TableRow[];
    getSettings(): T;
}

export interface TextFilterSettings {
    filterTerm: string;
    filterProps: string[];
}

export enum TableType {
    STATIONS,
    DELIVERIES,
}

export type TreeStatus = "collapsed" | "expanded";

export interface EditHighlightingServiceData {
    favouriteProperties: TableColumn[];
    otherProperties: TableColumn[];
    propToValuesMap: PropToValuesMap;
    ruleListItems: RuleListItem[];
}

export interface RuleListItem {
    id: RuleId;
    name: string;
    color: Color | null;
    shape: NodeShapeType | null;
    showInLegend: boolean;
    disabled: boolean;
    autoDisabled: boolean;
    isAnonymizationRule: boolean;
    effElementsCountTooltip: string;
    effElementsCount: number;
    conflictCount: number;
    ruleType: RuleType | null;
}

export interface EditRuleCore {
    id: string;
    name: string;
    userDisabled: boolean;
    complexFilterConditions: ComplexFilterCondition[];
    isValid: boolean;
}

export type EditRule =
    | ColorAndShapeEditRule
    | ColorEditRule
    | LabelEditRule
    | InvEditRule;

export enum RuleType {
    INVISIBILITY,
    LABEL,
    COLOR_AND_SHAPE,
    COLOR,
}

export interface ColorAndShapeEditRule extends EditRuleCore {
    type: RuleType.COLOR_AND_SHAPE;
    showInLegend: boolean;
    color: Color | null;
    shape: NodeShapeType | null;
}

export interface ColorEditRule extends EditRuleCore {
    type: RuleType.COLOR;
    showInLegend: boolean;
    color: Color;
}

export interface SimpleLabelEditRule extends EditRuleCore {
    type: RuleType.LABEL;
    labelProperty: string | null;
}

export interface ComposedLabelEditRule extends EditRuleCore {
    type: RuleType.LABEL;
    labelParts: LabelPart[];
    labelPrefix: string;
}

export type LabelEditRule = SimpleLabelEditRule | ComposedLabelEditRule;

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface InvEditRule extends EditRuleCore {
    type: RuleType.INVISIBILITY;
}

export type StationEditRule =
    | ColorAndShapeEditRule
    | LabelEditRule
    | InvEditRule;
export type DeliveryEditRule = LabelEditRule | InvEditRule | ColorEditRule;

export type DeliveryRuleType = RuleType.LABEL | RuleType.COLOR;
export type StationRuleType = RuleType.LABEL | RuleType.COLOR_AND_SHAPE;

export type RuleId = string;

export interface EditHighlightingState<T extends EditRule> {
    dataServiceInputState: DataServiceInputState;
    editRules: T[];
}
