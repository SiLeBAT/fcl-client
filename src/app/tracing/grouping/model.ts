import {
    StationData, GroupType, GroupData,
    StationTracingSettings, Position, DataServiceInputState
} from '../data.model';

export interface SetStationGroupsPayload {
    groupSettings: GroupData[];
    stationTracingSettings: StationTracingSettings[];
    stationPositions: { [key: string]: Position };
    selectedStations: string[];
    invisibleStations: string[];
}

export interface GroupingChange {
    addGroups: GroupData[];
    removeGroups: GroupData[];
}

export interface GroupingState extends DataServiceInputState {
    stationPositions: { [key: string]: Position };
}

export interface LinkGroup {
    linkStation: StationData;
    linkedStations: {
        linkedStation: StationData;
        linkKeys: string[];
    }[];
}

export type GroupInfoFun = (linkGroup: LinkGroup, groupNumber: number) => { id: string; name: string; groupType: GroupType };

export interface IsolatedComponent {
    ids: string[];
    support: string[];
}
