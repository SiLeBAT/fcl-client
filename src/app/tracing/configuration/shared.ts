import {
    DeliveryHighlightingRule, HighlightingRule, LinePatternType,
    StationHighlightingRule, TableColumn, TableRow
} from '../data.model';
import { PropToValuesMap } from './configuration.model';
import {
    ColorAndShapeEditRule, ColorEditRule, InvEditRule,
    LabelEditRule, RowFilter, RuleType, SimpleLabelEditRule
} from './model';
import * as _ from 'lodash';

export type EditRuleOfType<T extends RuleType> =
    T extends RuleType.LABEL ? LabelEditRule :
    T extends RuleType.COLOR_AND_SHAPE ? ColorAndShapeEditRule :
    T extends RuleType.COLOR ? ColorEditRule :
    T extends RuleType.INVISIBILITY ? InvEditRule : never;

export function filterTableRows(rows: TableRow[], rowFilters: RowFilter<any>[]): TableRow[] {
    let filteredRows = rows;
    for (const rowFilter of rowFilters) {
        filteredRows = rowFilter.filter(filteredRows);
    }
    return filteredRows;
}

export function extractPropToValuesMap(tableRows: TableRow[], tableColumns: TableColumn[]): PropToValuesMap {
    const propToValuesMap: PropToValuesMap = {};
    for (const column of tableColumns) {
        const values: string[] = _.uniq(
            tableRows
                .map(r => r[column.id] as (string | number | boolean))
                .filter(v => v !== undefined && v !== null)
                .map(v => typeof v === 'string' ? v : '' + v)
        ).sort();
        propToValuesMap[column.id] = values;
    }

    propToValuesMap[''] = _.uniq([].concat(...Object.values(propToValuesMap))).sort();

    return propToValuesMap;
}

export function isSimpleLabelRule(editRule: LabelEditRule): editRule is SimpleLabelEditRule {
    return (editRule as SimpleLabelEditRule).labelProperty !== undefined;
}

export function createDefaultHRule(): Omit<HighlightingRule, 'id' | 'name'> {
    return {
        userDisabled: false,
        autoDisabled: false,
        showInLegend: false,
        color: null,
        invisible: false,
        adjustThickness: false,
        labelProperty: null,
        valueCondition: null,
        logicalConditions: [[]]
    };
}

export function createDefaultStatHRule(): Omit<StationHighlightingRule, 'id' | 'name'> {
    return {
        ...createDefaultHRule(),
        shape: null
    };
}

export function createDefaultDelHRule(): Omit<DeliveryHighlightingRule, 'id' | 'name'> {
    return {
        ...createDefaultHRule(),
        linePattern: LinePatternType.SOLID
    };
}
