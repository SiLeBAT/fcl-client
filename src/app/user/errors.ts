import { InvalidServerInputHttpErrorResponse } from '@app/core/errors';
import { ValidationError } from '@app/core/model';
import { CODE_TO_FIELD_MAP } from './consts/error-code-mappings.consts';
import { RegistrationCredentials } from './models/user.model';

export class InvalidRegistrationInput {
    private _errors: Partial<Record<keyof RegistrationCredentials, ValidationError[]>> = {};
    private _message: string | null = null;

    get errors(): Partial<Record<keyof RegistrationCredentials, ValidationError[]>> {
        return this._errors;
    }

    get message(): string | null {
        return this._message;
    }

    constructor(error: InvalidServerInputHttpErrorResponse) {
        this._message = error.message;
        this.mapErrors(error.errors);
    }

    private mapErrors(validationErrors: ValidationError[]): void {
        this._errors = {};
        validationErrors.forEach(error => {
            const field = CODE_TO_FIELD_MAP[error.code] as undefined | keyof RegistrationCredentials;
            if (field !== undefined) {
                const fieldErrors = this._errors[field];
                if (fieldErrors !== undefined) {
                    fieldErrors.push(error);
                } else {
                    this._errors[field] = [error];
                }
            }
        });
    }
}
