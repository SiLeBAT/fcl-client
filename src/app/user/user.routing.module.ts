import { NgModule } from "@angular/core";
import { Routes, RouterModule, mapToCanActivate } from "@angular/router";

import { LoginViewComponent } from "./presentation/login-view/login-view.component";
import { RegisterViewComponent } from "./presentation/register-view/register-view.component";
import { RecoveryViewComponent } from "./presentation/recovery-view/recovery-view.component";
import { ResetViewComponent } from "./presentation/reset-view/reset-view.component";
import { ActivateViewComponent } from "./presentation/activate-view/activate-view.component";
import { AdminActivateViewComponent } from "./presentation/admin-activate-view/admin-activate-view.component";
import { ProfileContainerComponent } from "./container/profile-container/profile-container.component";
import { environment } from "./../../environments/environment";
import { AuthGuard } from "./guards/auth.guard";
import { NewsActivateViewComponent } from "./presentation/news-activate-view/news-activate-view.component";

const userRoutes: Routes = [
    {
        path: "users",
        children: [
            {
                path: "profile",
                component: ProfileContainerComponent,
                canActivate: environment.serverless
                    ? undefined
                    : mapToCanActivate([AuthGuard]),
            },
            {
                path: "login",
                component: LoginViewComponent,
            },
            {
                path: "register",
                component: RegisterViewComponent,
            },
            {
                path: "recovery",
                component: RecoveryViewComponent,
            },
            {
                path: "reset/:id",
                component: ResetViewComponent,
            },
            {
                path: "activate/:id",
                component: ActivateViewComponent,
            },
            {
                path: "adminactivate/:id",
                component: AdminActivateViewComponent,
            },
            {
                path: "newsactivate/:id",
                component: NewsActivateViewComponent,
            },
        ],
    },
];

@NgModule({
    imports: [RouterModule.forChild(userRoutes)],
    exports: [RouterModule],
})
export class UserRoutingModule {}
