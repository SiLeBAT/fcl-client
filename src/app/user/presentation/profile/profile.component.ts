import {Component, Input} from '@angular/core';
import {TokenizedUser} from '../../models/user.model';

@Component({
  selector: 'fcl-profile',
  templateUrl: './profile.component.html',
  styleUrls: ['./profile.component.scss'],
})
export class ProfileComponent {
  @Input() currentUser: TokenizedUser;
}
