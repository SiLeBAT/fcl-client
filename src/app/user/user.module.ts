import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { PasswordStrengthMeterModule } from 'angular-password-strength-meter';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { SharedModule } from './../shared/shared.module';
import { CoreModule } from '../core/core.module';
import { MainPageModule } from '../main-page/main-page.module';
import { UserRoutingModule } from './user.routing.module';
import { ContentModule } from '@app/content/content.module';

import { STATE_SLICE_NAME, reducer } from './state/user.reducer';
import { UserEffects } from './state/user.effects';
import { LoginContainerComponent } from './container/login-container/login-container.component';
import { LoginComponent } from './presentation/login/login.component';
import { LoginViewComponent } from './presentation/login-view/login-view.component';
import { RegisterContainerComponent } from './container/register-container/register-container.component';
import { RegisterComponent } from './presentation/register/register.component';
import { RegisterViewComponent } from './presentation/register-view/register-view.component';
import { RecoveryContainerComponent } from './container/recovery-container/recovery-container.component';
import { RecoveryComponent } from './presentation/recovery/recovery.component';
import { RecoveryViewComponent } from './presentation/recovery-view/recovery-view.component';
import { ResetContainerComponent } from './container/reset-container/reset-container.component';
import { ResetComponent } from './presentation/reset/reset.component';
import { ResetViewComponent } from './presentation/reset-view/reset-view.component';
import { ActivateContainerComponent } from './container/activate-container/activate-container.component';
import { ActivateComponent } from './presentation/activate/activate.component';
import { ActivateViewComponent } from './presentation/activate-view/activate-view.component';
import { AdminActivateContainerComponent } from './container/admin-activate-container/admin-activate-container.component';
import { AdminActivateComponent } from './presentation/admin-activate/admin-activate.component';
import { AdminActivateViewComponent } from './presentation/admin-activate-view/admin-activate-view.component';
import { ProfileContainerComponent } from './container/profile-container/profile-container.component';
import { ProfileComponent } from './presentation/profile/profile.component';
import { GdprAgreementComponent } from './presentation/gdpr-agreement/gdpr-agreement.component';
import { NewsActivateContainerComponent } from './container/news-activate-container/news-activate-container.component';
import { NewsActivateComponent } from './presentation/news-activate/news-activate.component';
import { NewsActivateViewComponent } from './presentation/news-activate-view/news-activate-view.component';

@NgModule({
    imports: [
        CommonModule,
        ContentModule,
        ReactiveFormsModule,
        PasswordStrengthMeterModule,
        SharedModule,
        CoreModule,
        MainPageModule,
        UserRoutingModule,
        StoreModule.forFeature(STATE_SLICE_NAME, reducer),
        EffectsModule.forFeature([UserEffects]),
        FormsModule
    ],
    exports: [],
    declarations: [
        LoginContainerComponent,
        LoginComponent,
        LoginViewComponent,
        RegisterContainerComponent,
        RegisterComponent,
        RegisterViewComponent,
        RecoveryViewComponent,
        RecoveryComponent,
        RecoveryContainerComponent,
        ResetContainerComponent,
        ResetComponent,
        ResetViewComponent,
        ActivateContainerComponent,
        ActivateComponent,
        ActivateViewComponent,
        AdminActivateContainerComponent,
        AdminActivateComponent,
        AdminActivateViewComponent,
        ProfileContainerComponent,
        ProfileComponent,
        GdprAgreementComponent,
        NewsActivateContainerComponent,
        NewsActivateComponent,
        NewsActivateViewComponent
    ],
    schemas: [CUSTOM_ELEMENTS_SCHEMA]

})
export class UserModule { }
