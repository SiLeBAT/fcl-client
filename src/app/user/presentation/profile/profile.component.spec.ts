import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { DebugElement, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';

import { ProfileComponent } from './profile.component';
import { MaterialModule } from '../../../shared/material.module';
import { SharedModule } from '../../../shared/shared.module';
import { TokenizedUser } from '../../models/user.model';
import { By } from '@angular/platform-browser';

describe('ProfileComponent', () => {

    let component: ProfileComponent;
    let fixture: ComponentFixture<ProfileComponent>;
    let debugElement: DebugElement;

    beforeEach(async(() => {
        // eslint-disable-next-line @typescript-eslint/no-floating-promises
        TestBed.configureTestingModule({
            declarations: [
                ProfileComponent
            ],
            imports: [
                MaterialModule,
                SharedModule,
                RouterTestingModule
            ],
            schemas: [CUSTOM_ELEMENTS_SCHEMA]
        }).compileComponents()
            .then(() => {
                fixture = TestBed.createComponent(ProfileComponent);
                component = fixture.componentInstance;
                debugElement = fixture.debugElement;
            });
    }));

    it('should create the component', () => {
        expect(component).toBeTruthy();
    });

    it('should render the component correctly', () => {
        expect(fixture).toMatchSnapshot();
    });

    it('should display the current user', () => {
        const currentUser: TokenizedUser = {
            email: 'email',
            firstName: 'first',
            lastName: 'last',
            token: 'test',
            gdprAgreementRequested: false
        };
        component.currentUser = currentUser;
        fixture.detectChanges();

        expect(fixture.nativeElement.outerHTML).toContain('first last');
        expect(fixture).toMatchSnapshot();
    });
});
