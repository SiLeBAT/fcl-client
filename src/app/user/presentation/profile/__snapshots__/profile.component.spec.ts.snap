// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileComponent should display the current user 1`] = `
<fcl-profile
  currentUser={[Function Object]}
>
  <fcl-single-center-card-layout
    cardtitle="User Profile"
  >
    <div
      class="fcl-single-center-card-layout-item fcl-single-center-card-layout-container"
    >
      <mat-card
        class="fcl-mat-card mat-card mat-focus-indicator"
      >
        <mat-card-header
          class="fcl-mat-card__header mat-card-header"
        >
          <div
            class="mat-card-header-text"
          >
            <mat-card-title
              class="mat-card-title"
            >
              <div
                class="mat-headline"
              >
                User Profile
              </div>
              <mat-divider
                aria-orientation="horizontal"
                class="mat-divider mat-divider-horizontal"
                role="separator"
              />
            </mat-card-title>
          </div>
        </mat-card-header>
        <mat-card-content
          class="mat-card-content"
        >
          <fcl-mat-card-body>
            <div
              class="fcl-mat-card-text"
            >
              <p>
                first last
              </p>
              <p>
                email
              </p>
            </div>
          </fcl-mat-card-body>
        </mat-card-content>
        <mat-card-actions
          class="fcl-mat-card-actions mat-card-actions"
        />
      </mat-card>
    </div>
  </fcl-single-center-card-layout>
</fcl-profile>
`;

exports[`ProfileComponent should render the component correctly 1`] = `
<fcl-profile>
  <fcl-single-center-card-layout
    cardtitle="User Profile"
  >
    <div
      class="fcl-single-center-card-layout-item fcl-single-center-card-layout-container"
    >
      <mat-card
        class="fcl-mat-card mat-card mat-focus-indicator"
      >
        <mat-card-header
          class="fcl-mat-card__header mat-card-header"
        >
          <div
            class="mat-card-header-text"
          >
            <mat-card-title
              class="mat-card-title"
            >
              <div
                class="mat-headline"
              >
                
              </div>
              <mat-divider
                class="mat-divider"
                role="separator"
              />
            </mat-card-title>
          </div>
        </mat-card-header>
        <mat-card-content
          class="mat-card-content"
        >
          <fcl-mat-card-body>
            <div
              class="fcl-mat-card-text"
            >
              <p>
                
              </p>
              <p>
                
              </p>
            </div>
          </fcl-mat-card-body>
        </mat-card-content>
        <mat-card-actions
          class="fcl-mat-card-actions mat-card-actions"
        />
      </mat-card>
    </div>
  </fcl-single-center-card-layout>
</fcl-profile>
`;
